tests/MongoDB.Driver.Tests/AuthenticationTests.cs

```diff
@@ -347,6 +352,16 @@ private void AssertAuthenticationSucceeds(MongoClientSettings settings, bool asy
                 {
                     client.ListDatabaseNames().ToEnumerable().ToList();
                 }
+                if (shouldSpeculativelyAuthenticateIfPossible &&
```

Comment:
Note: this technically adds a speculative auth test to all the other tests in this file except for the `Authentication_succeeds_when_user_has_Scram_Sha_1_Mechanism_and_mech`